File: Learnify.Service/Implementation/LessonService.cs
using Learnify.Domain.DTO;
using Learnify.Domain.Entities;
using Learnify.Repository.Implementation;
using Learnify.Repository.Interfaces;
using Learnify.Service.Interfaces;
using Learnify.Service.Mappers;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;

namespace Learnify.Service.Implementation
{
    public class LessonService : ILessonService
    {

        private readonly ILessonRepository lessonRepository;

        public LessonService(ILessonRepository lessonRepository)
        {
            this.lessonRepository = lessonRepository;
        }

        public Task DeleteLesson(Guid lessonId)
        {
            var lesson = lessonRepository.Get(lessonId);

            lessonRepository.Delete(lesson);

            return Task.CompletedTask;
        }


        public Lesson GetLesson(Guid lessonId)
        {
            var lesson = lessonRepository.Get(lessonId);

            return lesson;
        }

        public List<Lesson> GetLessons(Guid moduleId)
        {
            var lessons = lessonRepository.GetLessons(moduleId).ToList();

            return lessons;
        }

        public Task InsertLesson(InsertLesson insertLesson)
        {
            var lesson = LessonMapper.CopyFromDto(insertLesson);

            lessonRepository.Insert(lesson);

            return Task.CompletedTask;
        }

        public Task UpdateLesson(UpdateLesson updateLesson)
        {
            var lesson = LessonMapper.CopyFromUpdateDto(updateLesson);

            lessonRepository.Update(lesson);

            return Task.CompletedTask;
        }
    }
}

File: Learnify.UI/src/services/LessonService.js
const endpoint = '/lessons';

export const insertLesson = async (insertLesson) => {
    try {
        const response = await axios_config.post(endpoint, insertLesson);

        return response.status === 200 ? response.data : null;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const getLessonById = async (lessonId) => {
    try {
        const response = await axios_config.get(`${endpoint}/${lessonId}/get`);

        return response.status === 200 ? response.data : null;
    } catch (err) {
        console.error(`Error during the response: ${err.response}`);

        throw err;
    }
};

export const getLessons = async (moduleId) => {
    try {
        const response = await axios_config.get(`${endpoint}/${moduleId}`);

        return response.status === 200 ? response.data : null;
    } catch (err) {
        console.log(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const updateLesson = async (updateLesson) => {
    try {
        const response = await axios_config.put(endpoint, updateLesson);

        return response.status === 200;
    } catch (err) {
        console.error(`Error during the request: ${err.response}`);

        throw err;
    }
};

export const deleteLesson = async (lessonId) => {
    try {
        const response = await axios_config.delete(`${endpoint}/${lessonId}`);

        return response.status === 200;
    } catch (err) {
        console.log(`Error during the request: ${err.response}`);

        throw err;
    }
}
